src/main/java/nl/b3p/tailormap/api/controller/MapController.java

```diff
@@ -105,11 +114,14 @@ public ResponseEntity<Serializable> get(
                     Long appId) {
         logger.trace("Requesting 'map' for application id: " + appId);
 
-        // this could throw EntityNotFound, which is handles by handleEntityNotFoundException
-        // and in a normal flow this should not happen
-        // as appId is (should be) validated by calling the /app/ endpoint
-        Application application = applicationRepository.getReferenceById(appId);
-        if (application.isAuthenticatedRequired() && !AuthUtil.isAuthenticatedUser()) {
+        Application application = applicationRepository.findWithGeoservicesById(appId);
+        if (application == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
```

Comment:
Ik zou hier een `throw new EntityNotFoundException(...)` gebruiken zodat het bestaande mechanisme van de ExceptionHandler in  `public ErrorResponse handleEntityNotFoundException(EntityNotFoundException exception) ` op https://github.com/B3Partners/tailormap-api/blob/a08ff5cebab53a7121ef6699ba5dfaae8d675967/src/main/java/nl/b3p/tailormap/api/controller/MapController.java#L142 wordt gebruikt

Nu is er duplicatie van de foutmelding waarbij er 1 wel wordt gelogd en de andere niet...